### PR TITLE
AUTO-165: MOSK functional run failure analysis and fixes - stable/5.0

### PR DESCRIPTION
**Pull request by SuhasTA4**

### CODE REVIEW

**Comment by savitha-p** [comment]
Please attach local test run results for these tests

**Comment by SuhasTA4** [comment]
Attaching tempest.log snippet for each test showing test case PASS after handling scenarios like, 
- snapshot delete
- workload delete
- restore delete
image

File: tempest/api/workloadmgr/workload/test_workload.py [hunk 1]
```diff
@@ -324,16 +324,36 @@ def test_7_delete_workload(self):
...
-                    if (str(wc) == "deleted"):
+                    if (str(wc) == "None"):
                         reporting.add_test_step(
                             "Verification with DB", tvaultconf.PASS)
-                        LOG.debug("Workload successfully deleted")
+                        LOG.debug("Workload successfully deleted already. Returned None.")
                         self.deleted = True
                         break
+                    elif (str(wc) == "deleted"):
+                        LOG.debug("Returned value deleted for workload deletion status. Not Expected return value.")
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload.py]: This should be logged as error.

File: tempest/api/workloadmgr/workload/test_workload.py [hunk 2]
```diff
@@ -324,16 +324,36 @@ def test_7_delete_workload(self):
...
+                        self.deleted = False
+                        break
+                    elif (str(wc) == "error"):
+                        LOG.debug("Returned value error for workload deletion status. Failed to delete Workload")
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload.py]: This should be logged as error.

File: tempest/api/workloadmgr/snapshot/test_snapshots.py [hunk 3]
```diff
@@ -228,6 +228,9 @@ def test_4_delete_snapshot(self):
             if(str(wc) == "1"):
                 reporting.add_test_step("Verification", tvaultconf.PASS)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/snapshot/test_snapshots.py]: If 1 is returned, verification should be marked as fail.

File: tempest/api/workloadmgr/workload/test_workload.py [hunk 4]
```diff
@@ -324,16 +324,36 @@ def test_7_delete_workload(self):
                 LOG.debug("Workload successfully deleted")
                 self.deleted = True
             else:
-                while (str(wc) != "deleted"):
-                    time.sleep(5)
+                max_retries = 20
+                retry_count = 0
+                while (str(wc) != "deleted" or str(wc) != "None" or str(wc) != "error"):
+                    time.sleep(30)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload.py]: Reduce the sleep time

File: tempest/api/workloadmgr/restore/test_tvault1290_delete_restore.py [hunk 5]
```diff
@@ -68,18 +68,31 @@ def test_tvault1290_delete_restore(self):
             wc = query_data.get_workload_snapshot_status(
                 tvaultconf.snapshot_name, tvaultconf.snapshot_type_full, self.snapshot_id)
             LOG.debug("Workload snapshot status: " + str(wc))
-            while (str(wc) != "available" or str(wc) != "error"):
-                time.sleep(5)
+            max_retries = 20
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1290_delete_restore.py]: Add max_retries to tvaultconf.py and use that in the script.

File: tempest/api/workloadmgr/workload/test_workload.py [hunk 6]
```diff
@@ -324,16 +324,36 @@ def test_7_delete_workload(self):
                 LOG.debug("Workload successfully deleted")
                 self.deleted = True
             else:
-                while (str(wc) != "deleted"):
-                    time.sleep(5)
+                max_retries = 20
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload.py]: Add max_retries to tvaultconf.py and use that in the script.

File: tempest/api/workloadmgr/restore/test_tvault1290_delete_restore.py [hunk 7]
```diff
@@ -119,9 +132,13 @@ def test_tvault1290_delete_restore(self):
 
             wc = query_data.get_snapshot_restore_delete_status(
                 tvaultconf.restore_name, tvaultconf.restore_type)
+            LOG.debug("Snapshot delete status: " + str(wc))
             if (str(wc) == "1"):
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1290_delete_restore.py]: If 1 status is returned, then verification should fail

File: tempest/api/workloadmgr/restore/test_tvault1040_oneclick_restore.py [hunk 8]
```diff
@@ -92,8 +92,10 @@ def test_tvault1040_oneclick_restore(self):
             wc = query_data.get_snapshot_restore_status(
                 tvaultconf.restore_name, self.snapshot_id)
             LOG.debug("Snapshot restore status: " + str(wc))
+            max_retries = 20
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1040_oneclick_restore.py]: Add max_retries to tvaultconf.py and use that in the script.

File: tempest/api/workloadmgr/restore/test_tvault1040_oneclick_restore.py [hunk 9]
```diff
@@ -92,8 +92,10 @@ def test_tvault1040_oneclick_restore(self):
             wc = query_data.get_snapshot_restore_status(
                 tvaultconf.restore_name, self.snapshot_id)
             LOG.debug("Snapshot restore status: " + str(wc))
+            max_retries = 20
+            retry_count = 0
             while (str(wc) != "available" or str(wc) != "error"):
-                time.sleep(5)
+                time.sleep(30)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1040_oneclick_restore.py]: Reduce sleep time to may be 15sec

File: tempest/api/workloadmgr/restore/test_tvault1040_oneclick_restore.py [hunk 10]
```diff
@@ -103,9 +105,19 @@ def test_tvault1040_oneclick_restore(self):
                         "Snapshot one-click restore verification with DB", tvaultconf.PASS)
                     self.created = True
                     break
+                elif (str(wc) == "error"):
+                    LOG.debug("Failed to restore snapshot")
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1040_oneclick_restore.py]: This should be logged as error

File: tempest/api/workloadmgr/restore/test_tvault1037_list_restore.py [hunk 11]
```diff
@@ -81,18 +81,30 @@ def test_tvault1037_list_restore(self):
...
+                time.sleep(30)
                 wc = query_data.get_snapshot_restore_status(
                     tvaultconf.restore_name, self.snapshot_id)
                 LOG.debug("Snapshot restore status: " + str(wc))
                 if (str(wc) == "available"):
                     LOG.debug("Snapshot Restore successfully completed")
                     self.created = True
                     break
+                elif (str(wc) == "error"):
+                    LOG.debug("Failed to restore snapshot.")
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1037_list_restore.py]: This should be logged as error

File: tempest/api/workloadmgr/restore/test_tvault1037_list_restore.py [hunk 12]
```diff
@@ -81,18 +81,30 @@ def test_tvault1037_list_restore(self):
             wc = query_data.get_snapshot_restore_status(
                 tvaultconf.restore_name, self.snapshot_id)
             LOG.debug("Snapshot restore status: " + str(wc))
+            max_retries = 20
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1037_list_restore.py]: Add max_retries to tvaultconf.py and use that in the script.

File: tempest/api/workloadmgr/restore/test_tvault1037_list_restore.py [hunk 13]
```diff
@@ -81,18 +81,30 @@ def test_tvault1037_list_restore(self):
             wc = query_data.get_snapshot_restore_status(
                 tvaultconf.restore_name, self.snapshot_id)
             LOG.debug("Snapshot restore status: " + str(wc))
+            max_retries = 20
+            retry_count = 0
             while (str(wc) != "available" or str(wc) != "error"):
-                time.sleep(5)
+                time.sleep(30)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_tvault1037_list_restore.py]: Reduce sleep time to 15sec

**Review by savitha-p — CHANGES_REQUESTED** [review]
Comments added

**Review by savitha-p — APPROVED** [review]

**Review by priyankagupte15 — APPROVED** [review]

**Review by nasharatm — APPROVED** [review]

**Pull request merged**